### DISCUSSION TITLE
Pleas Help!, need any suggestions how improvement delete speed

### DISCUSSION THREAD

**Discussion by yazynin** [apache/ozone#5528] [General]
Hello, need suggestions how I can improvement delete speed, now disk space cleaned very slow, I used default settings for ozone-default.xml
----------------
Update:
In our case, setting the following values helped:
```
<property><name>ozone.scm.keyvalue.container.deletion-choosing.policy</name>
<value>org.apache.hadoop.ozone.container.common.impl.RandomContainerDeletionChoosingPolicy</value></property>
<property><name>hdds.datanode.block.deleting.limit.per.interval</name><value>50000</value></property>
```

**Comment by netscrol2016** [comment]
I have the same problem, the space will soon run out, and the space will not be freed up after deleting.

**Comment by errose28** [comment]
What is your Ozone version? There have been many improvements to deletion between 1.3.0 and master.

**Reply by yazynin** [reply]
1.3.0

**Reply by kerneltime** [reply]
I would recommend to try master, there have been many fixes since 1.3 and we are in the process of releasing 1.4

**Reply by yazynin** [reply]
Thank you for your answer, now we have around 500TB of data without the possibility to backup it, how safe it is to upgrade?
Or/and we should backup metadata etc for can unroll upgrade?

**Reply by DaveTeng0** [reply]
in case you need the steps to upgrade, here are the steps~ https://ozone.apache.org/docs/1.3.0/feature/nonrolling-upgrade.html
and remember to backup your original artifacts/jar before upgrade

**Reply by DaveTeng0** [reply]
btw, restoring the older versions is possible, until you 'finalizeupgrade'

**Comment by errose28** [comment]
Block deletion is asynchronous. Background tasks collect the blocks to be deleted at OM, which are sent to SCM, and then finally to datanodes to free the disk space. The following fixes that have been committed to master since 1.3.0 may be relevant:

If you are using FSO buckets:
- HDDS-7541: Improves speed of background directory deletion when an FSO directory is deleted.
- HDDS-7592: Fixes a bug where multiple directories with the same name and parent are deleted and the earlier deletes (and their children) are orphaned.

These fixes relate to SCM processing deletions:
- HDDS-8508: In our clusters this is the one we saw causing the most problems until it was patched. The other SCM fixes listed here were more minor issues with counters.
- HDDS-7125
- HDDS-7126
- HDDS-7306

These fixes relate to datanodes processing deletions:
- HDDS-7156: For us, this was the biggest issue blocking deletes on the datanodes until it was fixed, although others reported the issues below as well.
- HDDS-7259
- HDDS-7302
